platforms/web/lib/useComposerModel.ts

```diff
@@ -68,23 +68,26 @@ export function useComposerModel(
 
             if (initialContent) {
                 try {
-                    setComposerModel(
-                        new_composer_model_from_html(
-                            initialContent,
-                            0,
-                            initialContent.length,
-                        ),
+                    const newModel = new_composer_model_from_html(
+                        initialContent,
+                        0,
+                        initialContent.length,
                     );
+                    setComposerModel(newModel);
 
                     if (editorRef.current) {
+                        // we need to use the rust model as the source of truth, to allow it to do things
+                        // like add attributes to mentions automatically
+                        const modelContent = newModel.get_content_as_html();
                         replaceEditor(
                             editorRef.current,
-                            initialContent,
+                            modelContent,
                             0,
-                            initialContent.length,
+                            modelContent.length,
                         );
                     }
                 } catch (e) {
+                    // if the initialisation fails, due to a parsing failure of the html, fallback to an empty composer
```

Comment:
maybe we can try to display a raw version of the text in that case ? (just an idea for later)

Reply:
Yes that is a good idea. The failure here will be if we have poorly formed HTML though, and I don't know of a simple way to translate poorly formed HTML into raw text. I imagine it would be hard because of the number of ways the HTML could be incorrect.